system-contracts/test/manual-tests/L2LegacyBridgeFixUpgrade.manual.ts

```diff
@@ -0,0 +1,129 @@
+  REAL_COMPLEX_UPGRADER_CONTRACT_ADDRESS,
+} from "../shared/constants";
+import { publishBytecode } from "../shared/utils";
+import type { Context } from "mocha";
+
+describe("L2LegacyBridgeFixUpgrade tests", function () {
+  let oldBridgedEthVersion: number;
+
+  const aliasedGovernanceAddress =
+    process.env.ALIASED_GOVERNANCE_ADDRESS ||
+    ethers.utils.getAddress(ethers.utils.hexlify(ethers.utils.randomBytes(20)));
+  const l1ChainId = process.env.L1_CHAIN_ID || 1;
+  const bridgedEthAssetId = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(
+      ["uint256", "address", "bytes32"],
+      [l1ChainId, REAL_L2_NATIVE_TOKEN_VAULT_ADDRESS, ethers.utils.hexZeroPad(ethers.utils.hexlify(1), 32)]
+    )
+  );
+
+  before(async () => {
+    // Getting the old Bridged ETH version, if applicable
+    const l2NativeTokenVault = await ethers.getContractAt("IL2NativeTokenVault", REAL_L2_NATIVE_TOKEN_VAULT_ADDRESS);
+    const bridgedEthAddress = await l2NativeTokenVault.tokenAddress(bridgedEthAssetId);
+    if (bridgedEthAddress !== ethers.constants.AddressZero) {
+      const storageValue = await ethers.provider.getStorageAt(bridgedEthAddress, ethers.constants.Zero);
+      oldBridgedEthVersion = Number(BigInt(storageValue) & BigInt(0xff));
+    }
+  });
+
+  describe("upgrade", function () {
+    before(async () => {
+      const forceDeployer = await ethers.getImpersonatedSigner(REAL_FORCE_DEPLOYER_ADDRESS);
+      const complexUpgrader = ComplexUpgraderFactory.connect(REAL_COMPLEX_UPGRADER_CONTRACT_ADDRESS, forceDeployer);
+
+      const l2LegacyBridgeFixUpgradeFactory = await ethers.getContractFactory("L2LegacyBridgeFixUpgrade");
+      await publishBytecode(l2LegacyBridgeFixUpgradeFactory.bytecode);
+      const proxyAdminFactory = await ethers.getContractFactory("ProxyAdmin");
+      await publishBytecode(proxyAdminFactory.bytecode);
+      // Creating a dummy deterministic address for deployment
+      const dummyDeployAddress = create2Address(
+        complexUpgrader.address,
+        ethers.utils.hexlify(hashBytecode(l2LegacyBridgeFixUpgradeFactory.bytecode)),
+        ethers.utils.formatBytes32String("L2LegacyBridgeFixUpgrade"),
+        "0x"
+      );
+
+      const forceDeployments = [
+        {
+          bytecodeHash: hashBytecode(l2LegacyBridgeFixUpgradeFactory.bytecode),
+          newAddress: dummyDeployAddress,
+          callConstructor: false,
+          value: ethers.constants.Zero,
+          input: "0x",
+        },
+      ];
+
+      await complexUpgrader.forceDeployAndUpgrade(
+        forceDeployments,
+        dummyDeployAddress,
+        l2LegacyBridgeFixUpgradeFactory.interface.encodeFunctionData("upgrade", [
+          aliasedGovernanceAddress,
+          bridgedEthAssetId,
+        ])
+      );
+    });
+
+    it("System contracts are owned by the correct governance address", async () => {
+      const bridgehub = await ethers.getContractAt("Ownable2Step", REAL_BRIDGEHUB_ADDRESS);
```

Comment:
Could you please read and double check (maybe just console.log) the state of these values before the state? It would help to evaluate how helpful the test is (e.g. if a network had the correct governance from the beginning, it means I should use a different network)